pkg/proxy/iptables/proxier.go

```diff
@@ -135,6 +135,37 @@ func (lkct LinuxKernelCompatTester) IsCompatible() error {
 const sysctlRouteLocalnet = "net/ipv4/conf/all/route_localnet"
 const sysctlBridgeCallIPTables = "net/bridge/bridge-nf-call-iptables"
 
+// Since converting probabilities (floats) to strings is expensive
+// and we are using only probabilities in the format of 1/n, we are
+// precomputing some number of those and cache for future reuse.
+var precomputedProbabilities []string
+
+func init() {
+	const initiallyPrecomputed = 1000
+	precomputedProbabilities = make([]string, 0, initiallyPrecomputed+1)
+	precomputeProbabilities(initiallyPrecomputed)
+}
+
+func precomputeProbabilities(numberOfPrecomputed int) {
+	if len(precomputedProbabilities) == 0 {
```

Comment:
I talked with Tim offline, and he said he is fine with making it a field of Proxier. Will change.